### PR TITLE
Mortuary stays listing

### PR DESCRIPTION
**Pull request by Tatemsa**
[1](https://github.com/uni2growcm/u2g-openhospital-gui/issues/1)

### CODE REVIEW

File: sql/load_demo_data.sql [hunk 1]
```diff
@@ -1911,7 +1912,6 @@ INSERT INTO `oh_grouppermission` (`GP_ID`, `GP_UG_ID_A`, `GP_P_ID_A`, `GP_ACTIVE
 INSERT INTO `oh_grouppermission` (`GP_ID`, `GP_UG_ID_A`, `GP_P_ID_A`, `GP_ACTIVE`, `GP_CREATED_BY`, `GP_CREATED_DATE`, `GP_LAST_MODIFIED_BY`, `GP_LAST_MODIFIED_DATE`) VALUES (314,'admin',169,'1',NULL,NULL,NULL,NULL);
 INSERT INTO `oh_grouppermission` (`GP_ID`, `GP_UG_ID_A`, `GP_P_ID_A`, `GP_ACTIVE`, `GP_CREATED_BY`, `GP_CREATED_DATE`, `GP_LAST_MODIFIED_BY`, `GP_LAST_MODIFIED_DATE`) VALUES (315,'admin',170,'1',NULL,NULL,NULL,NULL);
 INSERT INTO `oh_grouppermission` (`GP_ID`, `GP_UG_ID_A`, `GP_P_ID_A`, `GP_ACTIVE`, `GP_CREATED_BY`, `GP_CREATED_DATE`, `GP_LAST_MODIFIED_BY`, `GP_LAST_MODIFIED_DATE`) VALUES (316,'admin',171,'1',NULL,NULL,NULL,NULL);
-INSERT INTO `oh_grouppermission` (`GP_ID`, `GP_UG_ID_A`, `GP_P_ID_A`, `GP_ACTIVE`, `GP_CREATED_BY`, `GP_CREATED_DATE`, `GP_LAST_MODIFIED_BY`, `GP_LAST_MODIFIED_DATE`) VALUES (317,'laboratorist',96,'1',NULL,NULL,NULL,NULL);
```
Comment (ArnaudFofou) [inline sql/load_demo_data.sql]: please add this line

File: sql/load_demo_data.sql [hunk 2]
```diff
@@ -2981,10 +2981,10 @@ UNLOCK TABLES;
 
 LOCK TABLES `oh_medicaldsrtype` WRITE;
 /*!40000 ALTER TABLE `oh_medicaldsrtype` DISABLE KEYS */;
-INSERT INTO `oh_medicaldsrtype` (`MDSRT_ID_A`, `MDSRT_DESC`, `MDSRT_DELETED`, `MDSRT_CREATED_BY`, `MDSRT_CREATED_DATE`, `MDSRT_LAST_MODIFIED_BY`, `MDSRT_LAST_MODIFIED_DATE`, `MDSRT_ACTIVE`) VALUES ('D','Drugs','N',NULL,NULL,NULL,NULL,1);
-INSERT INTO `oh_medicaldsrtype` (`MDSRT_ID_A`, `MDSRT_DESC`, `MDSRT_DELETED`, `MDSRT_CREATED_BY`, `MDSRT_CREATED_DATE`, `MDSRT_LAST_MODIFIED_BY`, `MDSRT_LAST_MODIFIED_DATE`, `MDSRT_ACTIVE`) VALUES ('K','Chemical','N',NULL,NULL,NULL,NULL,1);
-INSERT INTO `oh_medicaldsrtype` (`MDSRT_ID_A`, `MDSRT_DESC`, `MDSRT_DELETED`, `MDSRT_CREATED_BY`, `MDSRT_CREATED_DATE`, `MDSRT_LAST_MODIFIED_BY`, `MDSRT_LAST_MODIFIED_DATE`, `MDSRT_ACTIVE`) VALUES ('L','Laboratory','N',NULL,NULL,NULL,NULL,1);
```
Comment (ArnaudFofou) [inline sql/load_demo_data.sql]: revert changes here

File: sql/load_demo_data.sql [hunk 3]
```diff
@@ -2994,21 +2994,21 @@ UNLOCK TABLES;
...
-INSERT INTO `oh_medicaldsrward` (`MDSRWRD_WRD_ID_A`, `MDSRWRD_MDSR_ID`, `MDSRWRD_IN_QTI`, `MDSRWRD_OUT_QTI`, `MDSRWRD_LT_ID_A`, `MDSRWRD_CREATED_BY`, `MDSRWRD_CREATED_DATE`, `MDSRWRD_LAST_MODIFIED_BY`, `MDSRWRD_LAST_MODIFIED_DATE`, `MDSRWRD_ACTIVE`, `MDSRWRD_LOCK`) VALUES ('F',368,20,0,'1352158066768470913',NULL,NULL,NULL,NULL,1, 0);
-INSERT INTO `oh_medicaldsrward` (`MDSRWRD_WRD_ID_A`, `MDSRWRD_MDSR_ID`, `MDSRWRD_IN_QTI`, `MDSRWRD_OUT_QTI`, `MDSRWRD_LT_ID_A`, `MDSRWRD_CREATED_BY`, `MDSRWRD_CREATED_DATE`, `MDSRWRD_LAST_MODIFIED_BY`, `MDSRWRD_LAST_MODIFIED_DATE`, `MDSRWRD_ACTIVE`, `MDSRWRD_LOCK`) VALUES ('F',404,10,0,'AUTO_F_404',NULL,NULL,NULL,NULL,1, 0);
-INSERT INTO `oh_medicaldsrward` (`MDSRWRD_WRD_ID_A`, `MDSRWRD_MDSR_ID`, `MDSRWRD_IN_QTI`, `MDSRWRD_OUT_QTI`, `MDSRWRD_LT_ID_A`, `MDSRWRD_CREATED_BY`, `MDSRWRD_CREATED_DATE`, `MDSRWRD_LAST_MODIFIED_BY`, `MDSRWRD_LAST_MODIFIED_DATE`, `MDSRWRD_ACTIVE`, `MDSRWRD_LOCK`) VALUES ('F',415,200,0,'AUTO_F_415',NULL,NULL,NULL,NULL,1, 0);
```
Comment (ArnaudFofou) [inline sql/load_demo_data.sql]: revert code here

File: sql/load_demo_data.sql [hunk 4]
```diff
@@ -6704,10 +6716,10 @@ UNLOCK TABLES;
 
 LOCK TABLES `oh_user` WRITE;
 /*!40000 ALTER TABLE `oh_user` DISABLE KEYS */;
-INSERT INTO `oh_user` (`US_ID_A`, `US_UG_ID_A`, `US_PASSWD`, `US_DESC`, `US_CREATED_BY`, `US_CREATED_DATE`, `US_LAST_MODIFIED_BY`, `US_LAST_MODIFIED_DATE`, `US_ACTIVE`, `US_FAILED_ATTEMPTS`, `US_ACCOUNT_LOCKED`, `US_LOCK_TIME`, `US_LAST_LOGIN`, `US_DELETED`) VALUES ('admin','admin','$2a$10$FI/PMO0oSHHosF2PX8l3QuB0DJepVfnynbLZ9Zm2711bF2ch8db2S','administrator',NULL,NULL,NULL,NULL,1,0,0,NULL,NULL, false);
```
Comment (ArnaudFofou) [inline sql/load_demo_data.sql]: revert code here

File: sql/load_demo_data.sql [hunk 5]
```diff
@@ -6730,10 +6742,10 @@ UNLOCK TABLES;
 
 LOCK TABLES `oh_usergroup` WRITE;
 /*!40000 ALTER TABLE `oh_usergroup` DISABLE KEYS */;
-INSERT INTO `oh_usergroup` (`UG_ID_A`, `UG_DESC`, `UG_CREATED_BY`, `UG_CREATED_DATE`, `UG_LAST_MODIFIED_BY`, `UG_LAST_MODIFIED_DATE`, `UG_ACTIVE`, `UG_DELETED`) VALUES ('admin','USER with all the privileges',NULL,NULL,NULL,NULL,1, false);
```
Comment (ArnaudFofou) [inline sql/load_demo_data.sql]: revert the code here

**Review by ArnaudFofou — APPROVED** [review]

File: src/main/java/org/isf/mortuarystays/manager/MortuaryStayManager.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+		if (code.isEmpty()) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.common.pleaseinsertacode.msg")));
+		}
+		if (code.length() > 11) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.common.thecodeistoolongmax1char.msg")));
+		}
+
+		if (name.isEmpty()) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.mortuarystays.pleaseinsertavalidname.msg")));
+		}
```
Comment (SteveGT96) [inline src/main/java/org/isf/mortuarystays/manager/MortuaryStayManager.java:129]: Avoid this kind of validation. If you really wand to validate thoes field there, then you could annotate your entity fields with Validation annotations and use an validatio factoctotry to validate the entire class.
Eg:
`
@Column(name = "MRTST_CODE")
@Size(min = 11, message = "angal.mortuarystays.*.msg")
private String code;

@Column(name = "MRTST_MIN_DAYS")
@Positive(message = "angal.mortuarystays.*.msg")
private String minDays;

// Later in your code

Set<ConstraintViolation<MortuaryStay>> violations = validator.validate(mortuaryStay);
//throw some error with the constraint violations, while translating them.
`

File: src/main/java/org/isf/mortuarystays/manager/MortuaryStayManager.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+		if (ioOperations.isCodePresent(code)) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.mortuarystays.codealreadyinuse.msg")));
+		}
```
Comment (SteveGT96) [inline src/main/java/org/isf/mortuarystays/manager/MortuaryStayManager.java]: Is this check necessary ?

File: src/test/java/org/isf/mortuarystays/manager/MortuaryStayManagerTest.java [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+import org.isf.utils.exception.OHException;
+import org.isf.utils.exception.OHServiceException;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
```
Comment (SteveGT96) [inline src/test/java/org/isf/mortuarystays/manager/MortuaryStayManagerTest.java]: Did you import OH styles for this repository and reformat all files you've modified using it ?

**Review by SteveGT96 — CHANGES_REQUESTED** [review]

File: src/main/java/org/isf/mortuarystays/manager/MortuaryStayManager.java (hunk 6 above)
Comment (ArnaudFofou) [reply src/main/java/org/isf/mortuarystays/manager/MortuaryStayManager.java:129]: I think we can live it like that. to use the same way like officiel version

**Review by SteveGT96 — APPROVED** [review]
Approved by @ArnaudFofou !

**Comment by ArnaudFofou** [comment]
LGTM. Thanks!

**Pull request merged**